lib/configSchema/index.test.js

```diff
@@ -0,0 +1,176 @@
+'use strict';
+
+const runServerless = require('../../tests/utils/run-serverless');
+const CLI = require('../classes/CLI');
+const fixtures = require('../../tests/fixtures');
+const expect = require('chai').expect;
+
+describe('#configSchema', () => {
+  after(fixtures.cleanup);
+
+  const cases = [
+    {
+      isValid: true,
+      description: 'config',
+      mutation: {},
+    },
+
+    {
+      isValid: false,
+      errorMessage: '.service.name should match pattern "^[a-zA-Z][0-9a-zA-Z-]+$"',
+      description: 'service name',
+      mutation: {
+        service: { name: '1-first-number-is-not-allowed' },
+      },
+    },
+    {
+      isValid: false,
+      errorMessage:
+        '.service.awsKmsKeyArn should match pattern "^arn:(aws[a-zA-Z-]*)?:kms:[a-z0-9-]+-\\d+:\\d{12}:[^\\s]+$',
+      description: 'service awsKmsKeyArn',
+      mutation: {
+        service: {
+          name: 'some-service',
+          awsKmsKeyArn: 'invalidArn',
+        },
+      },
+    },
+    {
+      isValid: true,
+      description: 'service awsKmsKeyArn',
+      mutation: {
+        service: {
+          name: 'some-service',
+          awsKmsKeyArn: 'arn:aws:kms:us-east-1:123456789012:key/some-hash',
+        },
+      },
+    },
```

Comment:
To improve performance, I would combine cases that can be cumulated, e.g. validation of `app`, `org` and some other top properties, could be done with one fixture, we just need to ensure that error message for each was output

Reply:
Good point. 

I've combined tests with `app`, `org`, `resources` and `custom` valid properties into one test.

However, ironically, that test would throw an error because having in config both `app` and `org` makes `runServerless` to throw an error indicating that the user is not logged in (any ideas how to log in during test?). I've made a comment in code that would hopefully make things clearer.

Reply:
Hmm.. you pointed an interesting issue, I oversaw. The `org` and `app` should be purely in capacity of https://github.com/serverless/enterprise-plugin.

Therefore what I suggest:
1. Do not define a schema for `org` and `app`
2. Do not freeze top level object properties schema, and introduce a method to define a top level properties, is `defineTopLevelProperty` a good name?

Then prior merging this in, I'll introduce a PR to https://github.com/serverless/enterprise-plugin/ which will extend the schema with `app` and `org` schemas

Reply:
I haven't used dashboard or any enterprise features much. As far as I can see, these suggestions make sense. It would be totally fine to extend schema related to `app`/`org` properties in the mentioned plugin. As for the proposed method name, it also seems to be a good choice.

So I've implemented what's been suggested in https://github.com/serverless/serverless/pull/7335/commits/7654139de7684b02631fce27010459db5f2313c5.

@medikoo let me know if anything additional should be done within the scope of this PR.